pkg/oc/clusteradd/components/openshift-logging/openshift-logging.go

```diff
@@ -0,0 +1,99 @@
+package openshift_logging
+
+import (
+	"encoding/base64"
+	"fmt"
+	"github.com/golang/glog"
+	"io/ioutil"
+	"net/url"
+	"path"
+
+	configapi "github.com/openshift/origin/pkg/cmd/server/apis/config"
+	configapilatest "github.com/openshift/origin/pkg/cmd/server/apis/config/latest"
+	kerrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/origin/pkg/oc/clusteradd/componentinstall"
+	"github.com/openshift/origin/pkg/oc/clusterup/coreinstall/kubeapiserver"
+	"github.com/openshift/origin/pkg/oc/clusterup/docker/dockerhelper"
+	"github.com/openshift/origin/pkg/oc/clusterup/manifests"
+	"github.com/openshift/origin/pkg/oc/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	olNamespace = "openshift-logging"
+)
+
+type OpenshiftLoggingComponentOptions struct {
+	InstallContext componentinstall.Context
+}
+
+func (c *OpenshiftLoggingComponentOptions) Name() string {
+	return "openshift-logging"
+}
+
+func (c *OpenshiftLoggingComponentOptions) Install(dockerClient dockerhelper.Interface) error {
```

Comment:
Yeah, this is the kind of logic I was hoping for:

 1. gather information about the cluster
 2. pick out the bits you need
 3. install your resources from manifests
 4. wait for expected results.